client/client.go

```diff
@@ -35,13 +52,26 @@ type ProviderChannel struct {
 type Blockchain struct {
 	client    *ethclient.Client
 	bcTimeout time.Duration
+	nonceFunc nonceFunc
 }
 
+type nonceFunc func(ctx context.Context, account common.Address) (uint64, error)
```

Comment:
Better use interface definition. Noncer.nonce or smth, easier to find implementations with IDE.

Reply:
Won't work in this case, i'd have to wrap ethereum client for this.